### ISSUE TITLE
Template inside sub-folder not found

### ISSUE THREAD

**Issue by Teebo** [nest-modules/mailer#733]
Version: `"@nestjs-modules/mailer": "^1.6.1",`

I have an email template folder structure like

-src
    - templates
         - user
              - welcome.ejs
              
 In my email send options, I am giving a template: `'user/welcome'`.

In the mailer module configuration, I have the template dir option set as 
`dir: __dirname + '/templates',`

My Nest CLI configuration has

```
  "compilerOptions": {
    "assets": [
      { "include": "**/*.ejs", "outDir": "dist/src" },
      { "include": "**/*.txt", "outDir": "dist/src" },
      { "include": "**/*.png", "outDir": "dist/src" }],
    "watchAssets": true
  }
```

However, the templates are not found.

Another issue, in the `dist` folder, the templates are not under `template ` they are only under `template/user` which means the `outDir` option above in the Nest config did not work.

How can I achieve having templates inside folders?

**Comment by iWatchYouFromAfar** [comment]
I can confirm the bug.

For example this template:

```
/src/modules/mail/templates/staff/confirmation.hbs
```

I use like this:

```
    await this.mailerService
      .sendMail({
        to: email,
        subject: 'Subj',
        template: path.join(__dirname, './templates', 'staff', 'confirmation'),
        context: {
          url,
        },
      })
```

nest-cli.json:

```
{
  "collection": "@nestjs/schematics",
  "sourceRoot": "src",
  "compilerOptions": {
    "assets": [
      {
        "include": "modules/mail/templates/**/*.hbs"
      },
      {
        "include": "locales/**/**"
      }
    ],
    "watchAssets": true
  }
}
```

With `"@nestjs-modules/mailer": "^1.6.0"` everything is ok!

**Comment by Wunax** [comment]
I had exactly the same problem with the new version 1.6.1, the way the templates are resolved changed with this commit https://github.com/nest-modules/mailer/commit/9bee48b0e630c71c3702b83747dc41152670947e it checks if the path starts with "./", I had to change my code from like this: 

mailer.module.ts :
```javascript
  template: {
    dir: path.join(__dirname, './templates'),
    adapter: new EjsAdapter(),
    options: {
      strict: true,
    },
  },
```
mailer.service.ts
```javascript
return this.mailerService.sendMail({
  to: email,
  subject: 'Reset your password',
  template: './reset-password',
  context: {
    firstName,
    token,
  },
});
```

to like this:

mailer.module.ts
```javascript
  template: {
    dir: path.join(__dirname, './templates'),
    adapter: new EjsAdapter(),
    options: {
      strict: true,
    },
  },
```
mailer.service.ts
```javascript
return this.mailerService.sendMail({
  to: email,
  subject: 'Reset your password',
  template: 'reset-password',
  context: {
    firstName,
    token,
  },
});
```

**Comment by morpheus-87** [comment]
This PR should fix this issue: #743
